framework/encode/vulkan_capture_manager.cpp

```diff
@@ -1889,6 +1889,59 @@ void VulkanCaptureManager::PreProcess_vkCreateSwapchain(VkDevice
     }
 }
 
+void VulkanCaptureManager::PostProcess_vkCreateSwapchainKHR(VkResult                        result,
+                                                            VkDevice                        device,
+                                                            const VkSwapchainCreateInfoKHR* pCreateInfo,
+                                                            const VkAllocationCallbacks*    pAllocator,
+                                                            VkSwapchainKHR*                 pSwapchain)
+{
+    GFXRECON_UNREFERENCED_PARAMETER(device);
+    GFXRECON_UNREFERENCED_PARAMETER(pAllocator);
+    GFXRECON_UNREFERENCED_PARAMETER(pSwapchain);
+
+    GFXRECON_ASSERT(pCreateInfo != nullptr);
+
+    if (pCreateInfo)
+    {
+        // Vulkan Spec: Upon calling vkCreateSwapchainKHR with an oldSwapchain that is not VK_NULL_HANDLE, any images
+        // from oldSwapchain that are not acquired by the application may be freed by the implementation, which may
+        // occur even if creation of the new swapchain fails.
+
+        // The capture layer needs to be conservative and treat these images as destroyed now because the implementation
+        // is free to destroy and reuse the image handles before the retired swapchain is destroyed.
+        if (pCreateInfo->oldSwapchain != VK_NULL_HANDLE)
+        {
+            auto old_wrapper =
+                vulkan_wrappers::GetWrapper<vulkan_wrappers::SwapchainKHRWrapper>(pCreateInfo->oldSwapchain);
+            old_wrapper->retired = true;
```

Comment:
Should there be an 
`if (!old_wrapper->retired)`
here?

What happens if `vkCreateSwapchainKHR()` is called with the same `oldSwapchain` twice?

Reply:
Passing the same `oldSwapchain` twice would be invalid

> VUID-VkSwapchainCreateInfoKHR-oldSwapchain-01933
> If oldSwapchain is not VK_NULL_HANDLE, oldSwapchain must be a non-retired swapchain associated with native window referred to by surface

We could add an `assert()` for that, what do you think?

Reply:
I didn't realize the oldSwapchain is retired even if `vkCreateSwapchainKHR()` fails, not an issue then